conformance/packages/conformance-tests/src/resolver/dnssec/rfc4035/section_3/section_3_2.rs

```diff
@@ -52,6 +52,36 @@ fn do_bit_not_set_in_request() -> Result<()> {
     Ok(())
 }
 
+// this ensures that even in the presence of a cached (answer+rrsig) we strip the dnssec records as
+// per the RFC
+#[test]
+fn on_do_0_query_strips_dnssec_records_even_if_it_cached_a_previous_do_1_query() -> Result<()> {
```

Comment:
we should also test DO=0 followed by DO=1. given that the resolver must always set DO=1 in its outgoing queries to nameservers then that sequence of queries should result in the second one being a cache hit. that test should go under section_4_5 so I'll add it after #2244 lands